pkg/providers/v1/instances.go

```diff
@@ -36,27 +37,16 @@ import (
 // awsInstanceRegMatch represents Regex Match for AWS instance.
 var awsInstanceRegMatch = regexp.MustCompile("^i-[^/]*$")
 
-// InstanceID represents the ID of the instance in the AWS API, e.g. i-12345678
-// The "traditional" format is "i-12345678"
-// A new longer format is also being introduced: "i-12345678abcdef01"
-// We should not assume anything about the length or format, though it seems
-// reasonable to assume that instances will continue to start with "i-".
-type InstanceID string
-
-func (i InstanceID) awsString() *string {
-	return aws.String(string(i))
-}
-
-// KubernetesInstanceID represents the id for an instance in the kubernetes API;
+// KubernetesProviderID represents the id for an instance in the kubernetes API;
 // the following form
 //   - aws:///<zone>/<awsInstanceId>
 //   - aws:////<awsInstanceId>
 //   - aws:///<zone>/fargate-<eni-ip-address>
 //   - <awsInstanceId>
-type KubernetesInstanceID string
+type KubernetesProviderID string
```

Comment:
This type doesn't seem very useful to me, can we get rid of it in favor of something like:
```
func ParseProviderID(providerID string) (AWSProviderID, error)
```
?

Reply:
Agree and makes sense. Will fix